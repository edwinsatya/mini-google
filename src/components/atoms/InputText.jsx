import { useMemo } from "react";

const InputText = ({ className, value, onChange, type, disabled }) => {
  const classDisabled = useMemo(() => {
    return "bg-gray-300 dark:bg-gray-800 hover:dark:bg-gray-700";
  }, []);

  return (
    <div className="w-full">
      <input
        className={`${className} w-full ${disabled ? classDisabled : ""}`}
        type={type}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        disabled={disabled}
      />
    </div>
  );
};

export default InputText;
