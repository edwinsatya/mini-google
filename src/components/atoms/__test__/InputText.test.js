import { render, screen } from "@testing-library/react";
import InputText from "../InputText.jsx";

describe("Render input text atom component", () => {
  it("Should be render without crashing", () => {
    render(<InputText />);
    const inputText = screen.getByTestId("input-text");
    expect(inputText).toBeTruthy();
  });
});
