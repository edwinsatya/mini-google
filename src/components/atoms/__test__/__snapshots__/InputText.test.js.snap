// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText Atom Component Matches inputText snapshoot 1`] = `
<div
  className="w-full"
>
  <input
    className="undefined w-full "
    data-testid="input-text"
    onChange={[Function]}
    type="text"
  />
</div>
`;
